frb_codegen/src/config/opts_parser.rs

```diff
@@ -172,12 +233,66 @@ pub fn config_parse(mut raw: RawOpts) -> Vec<Opts> {
                 wasm_enabled: wasm,
                 dart3,
                 inline_rust,
+                shared: false,
+                shared_rust_output_path: shared_rust_output_path.clone(),
+                shared_dart_output_path: shared_dart_output_path.clone(),
                 bridge_in_method,
                 keep_going,
                 extra_headers: extra_headers.clone(),
             }
         })
-        .collect()
+        .collect::<Vec<_>>();
+
+    // if shared API-block(config) is essential, generate and add it to vec
+    assert!(!regular_configs.is_empty());
+    let all_configs = if regular_configs.len() == 1 {
```

Comment:
nit: seems there are a few different ways that we check whether it is multi-block mode, e.g. the one here + the "raw.rust_input.len() == 1" above. maybe we should extract to a variable. then it is easy to read as well as refactor/maintain.